lib/ansible/modules/web_infrastructure/pmp.py

```diff
@@ -0,0 +1,296 @@
+        headers = {
+            'Content-Type': 'text/json',
+        }  # admin test
+
+        url = self.format('https://{HOST}:{PORT}/restapi/json/v1/resources?AUTHTOKEN={TOKEN}')
+        data = json.dumps(data, sort_keys=False)
+        params = urllib.urlencode({'INPUT_DATA': data})
+        r = open_url(url, method="POST", headers=headers, data=params, validate_certs=self.validate_certs, use_proxy=self.use_proxy)
+        res = json.loads(r.read())['operation']['result']
+        status = res['status']
+        if status == 'Failed':
+            raise RuntimeError(res['message'])
+        return status
+
+
+def get_password(auth_pmp, params):
+    pmp = PasswordManagerPro(auth_pmp)
+    password = pmp.getPassword(params['resource_name'], params['account_name'])
+    return {'status': 'OK', 'password': password}
+
+
+def create_host(auth_pmp, params):
+    pmp = PasswordManagerPro(auth_pmp)
+    status = pmp.createResource(params['resource_name'], params['account_name'], params['password'], params['owner'],
+                                group=params['group'])
+    return {'status': status}
+
+
+# Some parameters are required depending on the operation:
+OP_REQUIRED = dict(create_host=['resource_name', 'account_name', 'password', 'owner'],
+                   get_password=['resource_name', 'account_name'],
+                   )
+
+
+def main():
+    global module
+    module = AnsibleModule(
+        argument_spec=dict(
+            uri=dict(required=True),
+            port=dict(required=False, default="7272"),
+            operation=dict(choices=['get_password', 'create_host'],
+                           aliases=['command'], required=True),
+            token=dict(required=True, no_log=True),
+            resource_name=dict(),
+            account_name=dict(),
+            password=dict(no_log=True),
+            owner=dict(),
+            group=dict(),
+            use_proxy=dict(default=False),
+            validate_certs=dict(default=False),
+            timeout=dict(type='float', default=10),
+        ),
+        supports_check_mode=False
+    )
+
+    op = module.params['operation']
+    # Check we have the necessary per-operation parameters
+    missing = []
+    for parm in OP_REQUIRED[op]:
+        if not module.params[parm]:
+            missing.append(parm)
+    if missing:
+        module.fail_json(msg="Operation %s require the following missing parameters: %s" % (op, ",".join(missing)))
+
+    # Handle rest of parameters
+    pmp_auth = {
+        'HOST': module.params['uri'],
+        'PORT': module.params['port'],
+        'TOKEN': module.params['token'],
+    }
+
+    # Dispatch
+    try:
+        # Lookup the corresponding method for this operation. This is
+        # safe as the AnsibleModule should remove any unknown operations.
+        thismod = sys.modules[__name__]
+        method = getattr(thismod, op)
+
+        ret = method(pmp_auth, module.params)
```

Comment:
This looks needlessly generic.  why not just do:
``` python
if op == 'get_password':
    get_password([...])
elif op == 'create_host':
    create_host([..])
```

If you think you're going to have a lot of operations, you can have a lookup table:
```
OPERATIONS = {'get_password': get_password, 'create_host': create_host}
method = OPERATIONS[op]([...])
```

but for just a few operations I'm not sure I would do that... it obscures what the code is doing in any given situation to code it like that.

Reply:
My plan is to add more operations on  the future, such as destroy an account, add an extra account for a resource, destroy a host...

But at this moment I have implemented this.